src/e3/archive.py

```diff
@@ -20,7 +20,10 @@
 
 
 if TYPE_CHECKING:
-    from typing import Callable, List, Optional, Sequence
+    from typing import Callable, List, Literal, Optional, Sequence, Union
+
+    UnpackAutoRemoveDirType = Literal["auto"]
```

Comment:
This doesn't seem to be used in either this commit or the next one. But perhaps you meant to use in the declaration of RemoveRootDirType?